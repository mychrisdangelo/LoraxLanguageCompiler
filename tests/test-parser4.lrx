// Resulting Abstract Syntax Tree
//
// program: Ast.program =
//   ([],
//    [{Ast.fname = "main"; ret_type = Ast.Lrx_Atom Ast.Lrx_Int; formals = [];
//      locals =
//       [("a",
//         Ast.Lrx_Tree
//          {Ast.datatype = Ast.Lrx_Char; degree = Ast.Int_Literal 1})];
//      body =
//       [Ast.Expr (Ast.Assign (Ast.Id "a", Ast.String_Literal "hello, world"));
//        Ast.Expr (Ast.Call ("print", [Ast.Id "a"]));
//        Ast.Return (Ast.Int_Literal 0)]}])

int main() {
	string a;
	a = "hello, world";
	print(a);
	return 0;
}