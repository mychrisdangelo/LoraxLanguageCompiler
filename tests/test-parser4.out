int main()
{
tree <char>a(1);
a = "hello, world";
print(a);
return 0;
}
